File: backend_code/models/cartItemModel.js
import sequelize from "../config/db_config.js";
import { DataTypes } from "sequelize";

const items=sequelize.define('CartItems',{
   id:{
         type:DataTypes.INTEGER,
         autoIncrement:true,
         primaryKey:true
   },
   cartId:{   // add to cart which belogs to a specific user
    type:DataTypes.INTEGER,
     allowNull:false,
   },
  sellerId:{  //seller id of the product that is added to cart
    type:DataTypes.INTEGER,
     allowNull:false,
   },
   productId:{    //which product to be added
    type:DataTypes.INTEGER,
     allowNull:false,
   },
   quantity:{  //quatity of product ordered
    type:DataTypes.INTEGER,
     allowNull:false,
     defaultValue: 1,
   },
   price:{  //product.price * quatity
    type:DataTypes.INTEGER,
     allowNull:false,
   },
   shippingPrice:{  
    type:DataTypes.INTEGER,
     allowNull:false,
     defaultValue: 0,
   },
   status:{
    type:DataTypes.STRING, //order item is approved or rejectedd etc...
    allowNull:true
   }
   
},{
  timestamps:true
}
);

export default items;

File: lib/models/cartItemModel.dart
import 'ProductModel.dart';

class CartItems {
  int? id;
  int? cartId;
  int? productId;
  int? sellerId;
  int? shippingPrice;
  int? quantity;
  int? price;
  String? status;
  String? createdAt;
  String? updatedAt;
  ProductModel? product;

  CartItems({
    this.id,
    this.cartId,
    this.productId,
    this.quantity,
    this.price,
    this.sellerId,
    this.shippingPrice,
    this.status,
    this.createdAt,
    this.updatedAt,
    this.product,
  });

  CartItems.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    cartId = json['cartId'];
    productId = json['productId'];
    shippingPrice = json['shippingPrice'];
    sellerId = json['sellerId'];
    quantity = json['quantity'];
    price = json['price'];
    status=json['status'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
    product =
    json['product'] != null
        ? new ProductModel.fromJson(json['product'])
        : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['cartId'] = this.cartId;
    data['productId'] = this.productId;
    data['quantity'] = this.quantity;
    data['shippingPrice']=this.shippingPrice;
    data['sellerId']=this.sellerId;
    data['price'] = this.price;
    data['status'] = this.status;
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    if (this.product != null) {
      data['product'] = this.product!.toJson();
    }
    return data;
  }
}
